### PR TITLE
Changed TransactionID into a struct

### PR DESCRIPTION
**Pull request by xpromache**
What started as an innocent wish to have the transaction id show better in the logs (because one cannot implement fmt::Display for a tuple), turned out into a lot of changes.

But I think the TransactionID was important enought to deserve its own struct.

### CODE REVIEW

File: cfdp-core/src/transaction/config.rs [hunk 1]
```diff
@@ -7,11 +7,29 @@ use crate::{
...
     pdu::{
         CRCFlag, Condition, EntityID, FaultHandlerAction, FileSizeFlag, FileStoreRequest,
-        MessageToUser, SegmentedData, TransactionSeqNum, TransmissionMode,
+        MessageToUser, SegmentedData, TransactionSeqNum, TransmissionMode, VariableID,
     },
 };
 
-pub type TransactionID = (EntityID, TransactionSeqNum);
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct TransactionID(pub EntityID, pub TransactionSeqNum);
```
Comment (mkolopanis) [inline cfdp-core/src/transaction/config.rs:15]: I don't mind this being a named tuple, that's an easy conversion with existing code.
I could imagine using name fields too if we want access to be less ambiguous for the reader.

**Review by mkolopanis — COMMENTED** [review]
Thanks for making this change. Looks pretty good, there are a few places in the user module that Linux seems unhappy about (though those are going away with #30 too). This seems easier to get merged in before that PR though.

File: cfdp-core/src/transaction/config.rs (hunk 1 above)
Comment (mkolopanis) [reply cfdp-core/src/transaction/config.rs:15]: @xpromache any thoughts here? I like to merge this before #30.

Comment (xpromache) [reply cfdp-core/src/transaction/config.rs:15]: For me it's ok like it is, I don't feel an urgency to change it to use named fields. I could change it if you want but I don't have time right now, maybe in 4-5 hours.

Comment (mkolopanis) [reply cfdp-core/src/transaction/config.rs:15]: This is fine for me for now. We can always change it later too.

**Pull request merged**